lib/puppet/type/file/source.rb

```diff
@@ -291,45 +283,58 @@ def chunk_file_from_disk
       end
     end
 
-    def get_from_puppet_source(source_uri, content_uri, &block)
-      options = { :environment => resource.catalog.environment_instance }
-      if content_uri
-        options[:code_id] = resource.catalog.code_id
-        request = Puppet::Indirector::Request.new(:static_file_content, :find, content_uri, nil, options)
-      else
-        request = Puppet::Indirector::Request.new(:file_content, :find, source_uri, nil, options)
-      end
+    def get_from_content_uri_source(content_uri, &block)
+      environment = Puppet.lookup(:current_environment).to_s
```

Comment:
When the environment changes due to a fact that was pluginsync'ed, then `Puppet.lookup(:current_environment)` may be be correct. Need to instead use `environment: resource.catalog.environment_instance` as was done before. It seems like a bug that the configurer doesn't reset the `:current_environment`, but that's a separate issue.

Reply:
Filed as https://tickets.puppetlabs.com/browse/PUP-10308

Reply:
Updated the code to use the `resource.catalog.environment` like it did before, and I call `to_s` on that since the http client requires an environment string, not an `Environment` object.